model/map/src/main/java/org/keycloak/models/map/datastore/MapExportImportManager.java

```diff
@@ -441,7 +448,107 @@ public RealmModel importRealm(InputStream requestBody) {
            the realm name and if the realm is enabled. For that scenario, it would need to create all missing elements,
            which is done by firing an event to call the existing implementation in the RealmManager. */
 
-        return ImportRealmFromRepresentation.fire(session, rep);
+        String id = rep.getId();
+        if (id == null || id.trim().isEmpty()) {
+            id = KeycloakModelUtils.generateId();
+        } else {
+            ReservedCharValidator.validate(id);
+        }
+
+        RealmModel realm = session.realms().createRealm(id, rep.getRealm());
+        RealmModel currentRealm = session.getContext().getRealm();
+
+        try {
+            session.getContext().setRealm(realm);
+            ReservedCharValidator.validate(rep.getRealm());
+            realm.setName(rep.getRealm());
+
+            // setup defaults that will be overwritten by the import if present in the import
+            setupRealmDefaults(realm);
+
+            /// THIS IS THE REAL IMPORT ///
+            RepresentationToModel.importRealm(session, rep, realm, false);
```

Comment:
This is actually calling 
```java
session.getProvider(DatastoreProvider.class).getExportImportManager().importRealm(rep, newRealm, skipUserDependent);
```
Should this line be thus replaced with the following?
```suggestion
            importRealm(rep, newRealm, skipUserDependent);
```